src/mono/mono/metadata/lock-tracer.h

```diff
@@ -15,7 +15,6 @@ typedef enum {
 	InvalidLock = 0,
 	LoaderLock,
 	ImageDataLock,
-	DomainLock,
```

Comment:
I would like to see `MemoryManagedLock` in here.  And also a dotnet/runtime (.NET 6+) update to https://www.mono-project.com/docs/advanced/runtime/docs/thread-safety/#the-locking-hierarchy